### PR TITLE
Add savedBefore, savedAfter, and saveSummary query parameters to SourceHandler.get

### PR DESCRIPTION
**Pull request by dannygoldstein**
This PR adds three new options to `SourceHandler.get`:

          - in: query
            name: savedBefore
            nullable: true
            schema:
              type: string
            description: |
              Only return sources that were saved before this UTC datetime.
          - in: query
            name: savedAfter
            nullable: true
            schema:
              type: string
            description: |
              Only return sources that were saved after this UTC datetime.
          - in: query
            name: saveSummary
            nullable: true
            schema:
              type: boolean
            description: |
              Boolean indicating whether to only return the source save
              information in the response (defaults to false). The schema
              of this summary information is:
              ```
                  {
                    "group_id": 2,
                    "created_at": "2020-11-13T22:11:25.910271",
                    "saved_by_id": 1,
                    "saved_at": "2020-11-13T22:11:25.910271",
                    "requested": false,
                    "unsaved_at": null,
                    "modified": "2020-11-13T22:11:25.910271",
                    "obj_id": "16fil",
                    "active": true,
                    "unsaved_by_id": null
                  }
              ```

Fixes https://github.com/fritz-marshal/fritz/issues/99

@cfremling

### CODE REVIEW

**Comment by dannygoldstein** [comment]
Ah, this needs a migration...

**Comment by dannygoldstein** [comment]
Migration added

File: skyportal/tests/api/test_sources.py [hunk 1]
```diff
@@ -346,3 +346,70 @@ def test_source_notifications_unauthorized(
...
+    assert len(sources) == 1
+    source = sources[0]
+
+    assert source['obj_id'] == public_source.id
+    assert source['group_id'] == public_group.id
+
+    # check the datetime formatting is properly validated
+    status, data = api(
+        "GET",
+        f"sources?saveSummary=true&group_ids={public_group.id}&savedBefore=2020-104-01T00:00:01.2412",
```
Comment (dmitryduev) [inline skyportal/tests/api/test_sources.py:412]: 104?

**Review by dmitryduev — APPROVED** [review]
LGTM!

**Comment by dannygoldstein** [comment]
Yep that’s what’s being tested - this should (and does) cause the test to fail

________________________________

**Comment by dannygoldstein** [comment]
Er, the query to fail - the test passes

________________________________

File: skyportal/handlers/api/source.py [hunk 2]
```diff
@@ -235,6 +236,43 @@ def get(self, obj_id=None):
...
+                    "created_at": "2020-11-13T22:11:25.910271",
+                    "saved_by_id": 1,
+                    "saved_at": "2020-11-13T22:11:25.910271",
+                    "requested": false,
+                    "unsaved_at": null,
+                    "modified": "2020-11-13T22:11:25.910271",
+                    "obj_id": "16fil",
+                    "active": true,
+                    "unsaved_by_id": null
+                  }
```
Comment (acrellin) [inline skyportal/handlers/api/source.py:274]: Indicate that the response data will include a list of these dicts under `response_json['data']['sources']`

File: skyportal/tests/api/test_sources.py [hunk 3]
```diff
@@ -346,3 +346,70 @@ def test_source_notifications_unauthorized(
     assert data["message"].startswith(
         "Twilio Communication SMS API authorization error"
     )
+
+
+def test_token_user_source_summary(
```
Comment (acrellin) [inline skyportal/tests/api/test_sources.py:351]: Check that the data we expect to not be present (obj data) are indeed not present

**Review by acrellin — CHANGES_REQUESTED** [review]

File: skyportal/handlers/api/source.py (hunk 2 above)
Comment (dannygoldstein) [reply skyportal/handlers/api/source.py:274]: done

File: skyportal/tests/api/test_sources.py (hunk 3 above)
Comment (dannygoldstein) [reply skyportal/tests/api/test_sources.py:351]: done

File: skyportal/tests/api/test_sources.py [hunk 4]
```diff
@@ -346,3 +346,70 @@ def test_source_notifications_unauthorized(
...
+        f"sources?saveSummary=true&group_ids={public_group.id}",
+        token=view_only_token_two_groups,
+    )
+    assert status == 200
+    assert "sources" in data['data']
+    sources = data['data']['sources']
+
+    assert len(sources) == 1
+    source = sources[0]
+    assert 'obj' not in source
```
Comment (acrellin) [inline skyportal/tests/api/test_sources.py:368]: Obj wouldn't be present anyways, right? Each dict in `sources` would just have all the obj data (ra, dec, mag, magerr, etc)

Comment (dannygoldstein) [reply skyportal/tests/api/test_sources.py]: nice catch, thanks. fixed

**Review by acrellin — APPROVED** [review]
Looks great. Will merge when green / only unrelated failures

**Pull request merged**